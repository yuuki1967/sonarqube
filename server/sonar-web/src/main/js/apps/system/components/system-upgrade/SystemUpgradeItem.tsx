/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import SystemUpgradeIntermediate from './SystemUpgradeIntermediate';
import DateFormatter from '../../../../components/intl/DateFormatter';
import { SystemUpgrade } from '../../../../api/system';
import { translate, translateWithParameters } from '../../../../helpers/l10n';

interface Props {
  type?: string;
  systemUpgrades: SystemUpgrade[];
}

export default function SystemUpgradeItem({ type, systemUpgrades }: Props) {
  const lastUpgrade = systemUpgrades[0];
  return (
    <div className="system-upgrade-version">
      {type && (
        <h1 className="spacer-bottom">
          <strong>{type}</strong>
        </h1>
      )}
      <p>
        <FormattedMessage
          defaultMessage={translate('system.version_is_availble')}
          id="system.version_is_availble"
          values={{ version: <b>SonarQube {lastUpgrade.version}</b> }}
        />
      </p>
      <p className="spacer-top">{lastUpgrade.description}</p>
      <div className="big-spacer-top">
        <DateFormatter date={lastUpgrade.releaseDate} long={true}>
          {formattedDate => (
            <span>{translateWithParameters('system.released_x', formattedDate)}</span>
          )}
        </DateFormatter>
        {lastUpgrade.changeLogUrl && (
          <a
            className="spacer-left"
            href={lastUpgrade.changeLogUrl}
            rel="noopener noreferrer"
            target="_blank">
            {translate('system.release_notes')}
          </a>
        )}
      </div>
      <SystemUpgradeIntermediate className="spacer-top" upgrades={systemUpgrades.slice(1)} />
      <div className="big-spacer-top">
        <a
          className="button"
          download={`sonarqube-${lastUpgrade.version}.zip`}
          href={lastUpgrade.downloadUrl}
          target="blank">
          {translateWithParameters('system.download_x', lastUpgrade.version)}
        </a>
        <a
          className="spacer-left"
          href="https://redirect.sonarsource.com/doc/upgrading.html"
          target="_blank">
          {translate('system.how_to_upgrade')}
        </a>
      </div>
    </div>
  );
}
