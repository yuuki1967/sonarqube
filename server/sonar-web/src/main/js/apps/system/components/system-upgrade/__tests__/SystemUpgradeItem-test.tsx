/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import SystemUpgradeItem from '../SystemUpgradeItem';

const UPGRADES = [
  {
    version: '5.6.7',
    description: 'Version 5.6.7 description',
    releaseDate: '2017-03-01',
    changeLogUrl: 'changelogurl',
    downloadUrl: 'downloadurl',
    plugins: {}
  },
  {
    version: '5.6.6',
    description: 'Version 5.6.6 description',
    releaseDate: '2017-04-02',
    changeLogUrl: 'changelogurl',
    downloadUrl: 'downloadurl',
    plugins: {}
  },
  {
    version: '5.6.5',
    description: 'Version 5.6.5 description',
    releaseDate: '2017-03-01',
    changeLogUrl: 'changelogurl',
    downloadUrl: 'downloadurl',
    plugins: {}
  }
];

it('should display correctly', () => {
  const wrapper = shallow(<SystemUpgradeItem systemUpgrades={UPGRADES} />);
  expect(wrapper).toMatchSnapshot();
});

it('should display a badge', () => {
  const wrapper = shallow(<SystemUpgradeItem systemUpgrades={UPGRADES} type="LTS Version" />);
  expect(wrapper.find('h1').exists()).toBeTruthy();
});
