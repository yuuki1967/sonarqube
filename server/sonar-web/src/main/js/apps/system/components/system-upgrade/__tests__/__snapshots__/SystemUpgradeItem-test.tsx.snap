// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correctly 1`] = `
<div
  className="system-upgrade-version"
>
  <p>
    <FormattedMessage
      defaultMessage="system.version_is_availble"
      id="system.version_is_availble"
      values={
        Object {
          "version": <b>
            SonarQube 
            5.6.7
          </b>,
        }
      }
    />
  </p>
  <p
    className="spacer-top"
  >
    Version 5.6.7 description
  </p>
  <div
    className="big-spacer-top"
  >
    <DateFormatter
      date="2017-03-01"
      long={true}
    >
      <Component />
    </DateFormatter>
    <a
      className="spacer-left"
      href="changelogurl"
      rel="noopener noreferrer"
      target="_blank"
    >
      system.release_notes
    </a>
  </div>
  <SystemUpgradeIntermediate
    className="spacer-top"
    upgrades={
      Array [
        Object {
          "changeLogUrl": "changelogurl",
          "description": "Version 5.6.6 description",
          "downloadUrl": "downloadurl",
          "plugins": Object {},
          "releaseDate": "2017-04-02",
          "version": "5.6.6",
        },
        Object {
          "changeLogUrl": "changelogurl",
          "description": "Version 5.6.5 description",
          "downloadUrl": "downloadurl",
          "plugins": Object {},
          "releaseDate": "2017-03-01",
          "version": "5.6.5",
        },
      ]
    }
  />
  <div
    className="big-spacer-top"
  >
    <a
      className="button"
      download="sonarqube-5.6.7.zip"
      href="downloadurl"
      target="blank"
    >
      system.download_x.5.6.7
    </a>
    <a
      className="spacer-left"
      href="https://redirect.sonarsource.com/doc/upgrading.html"
      target="_blank"
    >
      system.how_to_upgrade
    </a>
  </div>
</div>
`;
